exasol_advanced_analytics_framework/udf_communication/local_discovery_socket.py

```diff
@@ -0,0 +1,39 @@
+import socket
+import time
+from typing import Optional
+
+from exasol_advanced_analytics_framework.udf_communication.ip_address import IPAddress, Port
+
+
+class LocalDiscoverySocket:
+
+    def __init__(self, port: Port):
+        self._port = port
+        self._broadcast_ip = IPAddress(ip_address="127.255.255.255")
+        self._udp_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        self._udp_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        self._udp_socket.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+        self._udp_socket.bind((self._broadcast_ip.ip_address, self._port.port))
+
+    def send(self, message: bytes):
+        self._udp_socket.sendto(message, (self._broadcast_ip.ip_address, self._port.port))
+
+    def recvfrom(self, timeout_in_seconds: float) -> bytes:
+        if timeout_in_seconds<0.0:
```

Comment:
```suggestion
        if timeout_in_seconds < 0.0:
```

Reply:
At least error message suggests that timeout_in_seconds must not be equal to 0.0